### DISCUSSION TITLE
NSS error -8178 SEC_ERROR_BAD_KEY

### DISCUSSION THREAD

**Discussion by pluszeroplus** [curl/curl#14348] [Q&A] [answered]
**HI.
We used openssl 1.1 before and we want to switch to openssl 3.0. I recompiled the libcurl but it report "POST(https://172.31.0.253:8548/XXXX/) failed for Problem with the local SSL certificate, ret: 58, code: 0, errorBuf: unable to load client key: -8178 (SEC_ERROR_BAD_KEY)" after I recompiled it.
What is it?

**Comment by bagder** [comment, marked as answer]
"NSS error" implies that **NSS** (the TLS library no longer supported by curl) was used, not OpenSSL.

**Reply by pluszeroplus** [reply]
Why cURL not use OpenSSL? Somethings Wrong with my compile process?
This is how I compiled libcurl.so:
./configure  --prefix="${ROOT_PATH}/open_source/curl/curl-8.8.0" --with-openssl="/usr1/openssl_install_dir"

**Reply by bagder** [reply]
`curl -V` tells you exactly what it uses. Making the curl tool by mistake use an older libcurl installed previously in the system is quite a common mistake.

**Reply by pluszeroplus** [reply]
bin/curl -V
curl 8.8.0 (x86_64-pc-linux-gnu) libcurl/8.8.0 OpenSSL/3.0.14 zlib/1.2.7
Release-Date: 2024-05-22
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTPS-proxy IPv6 Largefile libz NTLM SSL threadsafe TLS-SRP UnixSockets

**Reply by pluszeroplus** [reply]
I found the reason. 
the libcurl I used is different from the one in the system. But I only copies the libcurl.so and libcurl.so.4.8 under. So yes the programe links to the libcurl in the system:
check by using lsof
```
lsof -p 5497 | grep curl
my_programe 5497 gandalf  mem       REG     253,1   436648  532810 /usr/lib64/libcurl.so.4.3.0
```
and the ldd of my_programe

```
ldd  my_programe  | grep curl
libcurl.so.4 => /lib64/libcurl.so.4 (0x00007f6de1428000)
```
I fix it by adding the soft link :
 XXXX/custom/lib/libcurl.so.4 XXXXXXX/lib/libcurl.so 
thank you.
